core-bundle/src/Resources/contao/classes/Ajax.php

```diff
@@ -327,41 +327,25 @@ public function executePostActions(DataContainer $dc)
 					{
 						$this->strAjaxId = preg_replace('/.*_([0-9a-zA-Z]+)$/', '$1', Input::post('id'));
 
-						$objVersions = new Versions($dc->table, $this->strAjaxId);
-						$objVersions->initialize();
-
-						$this->Database->prepare("UPDATE " . $dc->table . " SET " . Input::post('field') . "='" . ((Input::post('state') == 1) ? 1 : 0) . "' WHERE id=?")->execute($this->strAjaxId);
-
-						$objVersions->create();
+						$row = $dc->getCurrentRecord($this->strAjaxId, $dc->table);
+						$row[Input::post('field')] = (Input::post('state') == 1) ? 1 : 0;
+						$dc->setCurrentRecordCache($row['id'], $dc->table, $row);
 
 						if (Input::post('load'))
 						{
 							throw new ResponseException($this->convertToResponse($dc->editAll($this->strAjaxId, Input::post('id'))));
 						}
-
-						if (($intLatestVersion = $objVersions->getLatestVersion()) !== null)
-						{
-							throw new ResponseException($this->convertToResponse('<input type="hidden" name="VERSION_NUMBER" value="' . $intLatestVersion . '">'));
-						}
 					}
 					else
 					{
-						$objVersions = new Versions($dc->table, $dc->id);
-						$objVersions->initialize();
-
-						$this->Database->prepare("UPDATE " . $dc->table . " SET " . Input::post('field') . "='" . ((Input::post('state') == 1) ? 1 : 0) . "' WHERE id=?")->execute($dc->id);
-
-						$objVersions->create();
+						$row = $dc->getCurrentRecord();
+						$row[Input::post('field')] = (Input::post('state') == 1) ? 1 : 0;
```

Comment:
I'm not sure this is correct. You are not supposed to fiddle around with the current record because that is supposed to represent the record as it currently is stored in the database. I think what you're looking for is setting this value to `$arrSubmit` but not call `submit()` yet. That way, I will still get the correct values when my listener calls `$dc->getCurrentRecord()`. See what I mean?

Reply:
What you're doing here is kind of introducing the active record pattern again which we certainly should avoid :)

Reply:
`$dc->getCurrentRecord()` never includes the submitted data. The palette rendering checks the current record data, and this is the easiest way to manipulate that. I agree its not beautiful, but then again it's `DC_Table` 😂

Reply:
But what if the palette rendering would do `$fieldValue = $this->arrSubmit[$field] ?? $this->getCurrentRecord()[$field]`?

Reply:
Or in other words: We want to put the DC in a state and render the palette for that state. We don't want to submit it, nor fiddle around with the current record. It's basically just `->save()` or `->row()` without submit 🤔